tests/pytests/functional/cli/test_api.py

```diff
@@ -0,0 +1,34 @@
+import os
+import sys
+import tempfile
+
+import pytest
+from salt.cli.api import SaltAPI
+from tests.support.mock import patch
+
+
+@pytest.mark.slow_test
+def test_start_shutdown():
+    pidfile = tempfile.mktemp()
+    logfile = tempfile.mktemp()
+    api = SaltAPI()
+    with pytest.raises(SystemExit):
+        try:
+            # testing environment will fail if we use default pidfile
+            # overwrite sys.argv so salt-api does not use testing args
+            with patch.object(
+                sys, "argv", [sys.argv[0], "--pid-file", pidfile, "--log-file", logfile]
+            ):
+                api.start()
+                assert os.path.isfile(pidfile)
+                assert os.path.isfile(logfile)
+                api.shutdown()
+        finally:
+            try:
+                os.remove(pidfile)
+            except OSError:
+                pass
+            try:
+                os.remove(logfile)
+            except OSError:
+                pass
```

Comment:
**suggestion** (non-blocking): Could add a contextmanager for this:

```
from contextlib import contextmanager

@contextmanager
def mktemp():
    fname = tempfile.mktemp()
    try:
        yield fname
    finally:
        try:
            os.remove(fname)
        except OSError:
            pass
```

Then it's just 

```
with mktemp() as pidfile, mktemp() as logfile, patch.object(...):
```

and the rest of this boilerplate disappears.